### DISCUSSION TITLE
Minimal separating sets: checking and construction

### DISCUSSION THREAD

**Discussion by adam2392** [networkx/networkx#5859] [Algorithms]
Relevant for the "causal" [submodule discussion](https://github.com/networkx/networkx/discussions/5811), we are many times interested in minimal separating sets in DAGs/undirected graphs. We would like to perform the following operations many times:

1. check that a set is minimal
2. construct a minimal separating set if it exists

These would be useful to add to networkx, possibly in conjunction with the `d_separated` algorithm. This is also a fairly standard algorithm at this point if concerned with some inclusion criterion.

Reference: https://ftp.cs.ucla.edu/pub/stat_ser/r254.pdf

**Comment by dschult** [comment]
We have [d_separation](https://networkx.org/documentation/stable/reference/algorithms/d_separation.html) for DAGs already, so I'm hoping that a version for MixedEdgeGraph would be easy to build from that.

I don't know the algorithms for checking if a separating set is minimal, nor constructing such a set. It looks like there are algorithms for it though... I'm guessing it would be fine to add those functions to NetworkX.

**Reply by adam2392** [reply]
I can add a few PRs to the `d_separation` submodule then:

1. compute a minimal separating set for a DAG
2. to check if a proposed set is a minimal separating set in a DAG

I have these algorithms implemented on my end. They all were relatively straightforward and would complement the `d_separated` function in networkx.

**Comment by adam2392** [comment]
This will be addressed by: https://github.com/networkx/networkx/pull/5898

**Comment by adam2392** [comment]
As a next step, I would propose adding these capabilities to the `m_separated` function present in https://github.com/networkx/networkx/pull/5947. This would further enhance the `MixedEdgeGraph` use-case and also add some more mixed-edge algorithms to the proposed `causal` submodule.
